File: bin/Data/AUI/Periodic/Sources/code_aimagewidget.cpp
// AImageWidget application source code
#include <Urho3D/AUI/AImageWidget.h>
#include "PeriodicApp.h"

void PeriodicApp::setup_aimagewidget( AWidget *layout )
{
    PODVector<AWidget*> dest;
    layout->SearchWidgetClass( "TBButton", dest );
    for (unsigned ii = 0; ii < dest.Size(); ii++)
        SubscribeToEvent(dest[ii], E_WIDGETEVENT, URHO3D_HANDLER(PeriodicApp, HandleAimagewidgetEvent ));
}

void PeriodicApp::HandleAimagewidgetEvent(StringHash eventType, VariantMap& eventData)
{
    using namespace WidgetEvent;
    AWidget* widget = static_cast<AWidget*>(eventData[P_TARGET].GetPtr());
    if ( widget == NULL ) return;
    if (eventData[P_TYPE] == UI_EVENT_TYPE_CLICK)
    {
        if (widget->GetId() == "uiimagewidgetcode" )
        {
            AppLog( "AImageWidget support : " + widget->GetId() + " was pressed " );
            ViewCode ( "Sources/code_aimagewidget.cpp", widget->GetParent() );
        }
        if (widget->GetId() ==  "uiimagewidgetlayout" )
        {
            AppLog( "AImageWidget support : " + widget->GetId() + " was pressed ");
            ViewCode ( "Scenes/layout_aimagewidget.ui.txt", widget->GetParent() );
        }

        if (widget->GetId() ==  "imagecolor" )
        {
            AppLog( "AImageWidget action : " + widget->GetId() + " was pressed ");
            AImageWidget *img1 = static_cast<AImageWidget*>( widget->FindWidget("imagewidgetdemo") );
            img1->SetImage("Textures/HSV21.png");
        }
        if (widget->GetId() ==  "imagenewbuild" )
        {
            AppLog( "AImageWidget action : " + widget->GetId() + " was pressed ");
            AImageWidget *img2 = static_cast<AImageWidget*>( widget->FindWidget("imagewidgetdemo") );
            img2->SetImage("Textures/newbuilddetected_header.jpg");
        }
    }
}

File: bin/Data/AUI/Periodic/LuaScripts/code_aimagewidget.lua
-- AImageWidget application source code


function setup_uiimagewidget( layout )
    local dest = layout:SearchAWidgetClass( "TBButton" )
    for i, awid in ipairs(dest) do
        SubscribeToEvent(awid, "WidgetEvent", "HandleUiimagewidgetEvent")
	end
end

function HandleUiimagewidgetEvent(eventType, eventData)
    local widget = eventData["Target"]:GetPtr("AWidget")
    if widget == nil then
 		return;
	end
    if eventData["Type"]:GetInt() == UI_EVENT_TYPE_CLICK then
        if widget:GetId() == "uiimagewidgetcode" then
            AppLog( "AImageWidget support : " .. widget:GetId() .. " was pressed " )
            ViewCode ( "LuaScripts/code_aimagewidget.lua", widget:GetParent())
        end
        if widget:GetId() ==  "uiimagewidgetlayout" then
            AppLog( "AImageWidget support : " .. widget:GetId() .. " was pressed ")
            ViewCode ( "Scenes/layout_aimagewidget.ui.txt", widget:GetParent())
        end

        if widget:GetId() ==  "imagecolor" then
            AppLog( "AImageWidget action : " .. widget:GetId() .. " was pressed ")
            local img1 = widget:FindWidget("imagewidgetdemo")
			if img1 ~= nil then 
 				local i2 = tolua.cast( img1, "AImageWidget")
            	i2:SetImage("Textures/HSV21.png")
			end
        end
        if widget:GetId() ==  "imagenewbuild" then
            AppLog( "AImageWidget action : " .. widget:GetId() .. " was pressed ")
            local img1 = widget:FindWidget("imagewidgetdemo")
			if img1 ~= nil then
 				local i2 = tolua.cast( img1, "AImageWidget")
           		i2:SetImage("Textures/newbuilddetected_header.jpg")
			end
        end
    end
end

